tests/system/test_root_metronome.py

```diff
@@ -239,6 +239,28 @@ def test_docker_job():
         assert len(client.get_runs(job_id)) == 1
 
 
+def test_metronome_shutdown_with_no_extra_tasks():
+    """ Test for METRONOME-100 regression
+        When Metronome is restarted it incorrectly started another task for already running job run task.
+    """
+    client = metronome.create_client()
+    job_id = "metronome-shutdown-{}".format(uuid.uuid4().hex)
+    with job(job_no_schedule(job_id)):
+        # run a job before we shutdown Metronome
+        client.run_job(job_id)
+        time.sleep(2)
+        common.assert_one_job_run_with_one_task(client, job_id)
+
+        # restart metronome process
+        metronome_leader = shakedown.marathon_leader_ip()
+        shakedown.run_command_on_agent(metronome_leader, 'sudo systemctl restart dcos-metronome')
+        shakedown.time_wait(lambda: common.metronome_available_predicate(), timeout_seconds=timedelta(minutes=1).total_seconds())
+
+        # verify that no extra job runs were started when Metronome was restarted
+        time.sleep(20) # launching a new task takes some time, we have to give marathon enough time to do that
```

Comment:
the way we manage this in marathon (which wasn't available in metronome until recently) is to check the deployment queue.   We should do that here for the `time.sleep(2)` and for the `time.sleep(20)`.     It is best if we add this call to the metronome class in dcos-cli... but that will take time to propagate for our use here.   What you want to do is review marathon.deployment_wait https://github.com/dcos/shakedown/blob/e711e691d9692b06137465da61f0b3c6f7a53131/shakedown/dcos/marathon.py#L70
for metronome hit /metronome/v1/queue and wait for the json return of `[]` (up until a timeout)

Reply:
I am not sure that deployment wait will solve this, because I think there might be a time, when metronome is up (API is returning HTTP 200) but executor actor did not manage to put anything into the queue...

regarding that queue we should definitely take advantage of that. I added a ticket for that yesterday

Reply:
This is one possibility. Querying `v1/jobs` like I posted above is another.